pkg/cleaner/cleaning.go

```diff
@@ -105,12 +109,26 @@ func (c *FileContentObfuscator) ObfuscateFile(inputFile string, outputFile strin
 	// we need to assess whether the file exists already to ensure we don't overwrite existing obfuscated data.
 	// that can happen while obfuscating file names and their paths.
 	// Additionally, the stat check is required because os.O_CREATE will implicitly os.O_TRUNC if a file already exist
+	c.pathCollisionMutex.Lock()
+	defer c.pathCollisionMutex.Unlock()
```

Comment:
this is great, the problem is that this now locks the whole obfuscation process below as well (which is what we want to actually parallelize). 

Can you pull the locking, your added logic + OpenFile into a dedicated method that is locked? that way the lock has a clear and minimal scope